app/main.py

```diff
@@ -1,3 +1,54 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+import datetime
+import os
+
+from app.data import create_clients, create_shops
+
+
+def shop_trip() -> None:
+    with open(os.path.join("app", "config.json"), "r") as json_file:
+        data = json.load(json_file)
+    fuel_price = data["FUEL_PRICE"]
+
+    clients = create_clients(data)
+    shops = create_shops(data)
+
+    for client_name, client_info in clients.items():
+        print(f"{client_name} has {client_info.money} dollars")
+
+        client_info.calculate_shopping_costs(shops, fuel_price)
+        min_shop_name, min_shop_cost = min(
+            client_info.shopping_costs.items(), key=lambda x: x[1]
+        )
+
+        if client_info.money < min_shop_cost:
+            print(f"{client_name} doesn't have enough money to "
+                  f"make a purchase in any shop")
+            return
+
+        print(f"{client_name} rides to {min_shop_name}")
+        print("")
```

Comment:
You can use print once and `\n` inside a string. Fix everywhere

Reply:
Fixed